pkg_resources/api_tests.txt

```diff
@@ -338,49 +338,72 @@ Environment Markers
     >>> import os
 
     >>> print(im("sys_platform"))
-    Invalid marker: 'sys_platform', parse error at ''
+    Expected marker operator, one of <=, <, !=, ==, >=, >, ~=, ===, in, not in
+        sys_platform
+                    ^
 
     >>> print(im("sys_platform=="))
-    Invalid marker: 'sys_platform==', parse error at ''
+    Expected a marker variable or quoted string
+        sys_platform==
+                      ^
 
     >>> print(im("sys_platform=='win32'"))
     False
 
     >>> print(im("sys=='x'"))
-    Invalid marker: "sys=='x'", parse error at "sys=='x'"
+    Expected a marker variable or quoted string
+        sys=='x'
+        ^
 
     >>> print(im("(extra)"))
-    Invalid marker: '(extra)', parse error at ')'
+    Expected marker operator, one of <=, <, !=, ==, >=, >, ~=, ===, in, not in
+        (extra)
+              ^
 
     >>> print(im("(extra"))
-    Invalid marker: '(extra', parse error at ''
+    Expected marker operator, one of <=, <, !=, ==, >=, >, ~=, ===, in, not in
+        (extra
+              ^
 
     >>> print(im("os.open('foo')=='y'"))
-    Invalid marker: "os.open('foo')=='y'", parse error at 'os.open('
+    Expected a marker variable or quoted string
+        os.open('foo')=='y'
+        ^
 
     >>> print(im("'x'=='y' and os.open('foo')=='y'"))   # no short-circuit!
-    Invalid marker: "'x'=='y' and os.open('foo')=='y'", parse error at 'and os.o'
+    Expected a marker variable or quoted string
+        'x'=='y' and os.open('foo')=='y'
+                     ^
 
     >>> print(im("'x'=='x' or os.open('foo')=='y'"))   # no short-circuit!
-    Invalid marker: "'x'=='x' or os.open('foo')=='y'", parse error at 'or os.op'
```

Comment:
In this test, the exception raised by the new version of `packaging` is "less user friendly", they report an expected error for a missing `'y'` key.